lib/puppet/provider/ec2_vpc_routetable/v2.rb

```diff
@@ -31,18 +31,20 @@ def self.prefetch(resources)
 
   def self.route_to_hash(region, route)
     ec2 = ec2_client(region)
-    gateway = if route.gateway_id == 'local'
-      'local'
-    else
-      begin
-        igw_response = ec2.describe_internet_gateways(internet_gateway_ids: [route.gateway_id])
-        name_from_tag(igw_response.data.internet_gateways.first)
-      rescue Aws::EC2::Errors::InvalidInternetGatewayIDNotFound
+    if route.state == 'active'
```

Comment:
We query for the route tables rather than explicitly for the routes. And although route tables support a filter for state it's misleading. Filtering for active will return all route tables (and routes for that table) with an active route. So a route table with one active and one black hole route will be returned and will have both routes, even if you filter at the route table level for active.